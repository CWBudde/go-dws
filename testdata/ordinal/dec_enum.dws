// Test Dec() with enum values
// Task 9.25: Dec() built-in function for enums

type TColor = (Red, Green, Blue, Yellow);

var c: TColor := Yellow;
PrintLn('Initial color ordinal: ', Ord(c));
PrintLn('Initial color value: ', Integer(c));

Dec(c);
PrintLn('After first Dec(): ', Ord(c));

Dec(c);
PrintLn('After second Dec(): ', Ord(c));

Dec(c);
PrintLn('After third Dec(): ', Ord(c));

// Test with enum with explicit values
type TStatus = (Ok = 0, Warning = 5, Error = 10);

var s: TStatus := Error;
PrintLn('Status initial: ', Ord(s));

Dec(s);
PrintLn('After Dec (Error to Warning): ', Ord(s));

Dec(s);
PrintLn('After Dec (Warning to Ok): ', Ord(s));
