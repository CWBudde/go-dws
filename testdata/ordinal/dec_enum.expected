Initial color ordinal:  3
Initial color value:  3
After first Dec():  2
After second Dec():  1
After third Dec():  0
Status initial:  10
After Dec (Error to Warning):  5
After Dec (Warning to Ok):  0
